dot/network/connmgr.go

```diff
@@ -73,18 +79,40 @@ func (*ConnManager) GetTagInfo(peer.ID) *connmgr.TagInfo { return &connmgr.TagIn
 // TrimOpenConns peer
 func (*ConnManager) TrimOpenConns(ctx context.Context) {}
 
-// Protect peer
-func (*ConnManager) Protect(peer.ID, string) {}
+// Protect peer will add the given peer to the protectedPeerMap wich will
+// protect the peer from pruning.
+func (cm *ConnManager) Protect(id peer.ID, tag string) {
+	cm.protectedPeerMapMu.Lock()
+	defer cm.protectedPeerMapMu.Unlock()
 
-// Unprotect peer
-func (*ConnManager) Unprotect(peer.ID, string) bool { return false }
+	cm.protectedPeerMap[id] = struct{}{}
+}
+
+// Unprotect peer will remove the given peer from prune protection.
+// returns true if we have successfully removed the peer from the
+// protectedPeerMap. False otherwise.
+func (cm *ConnManager) Unprotect(id peer.ID, tag string) bool {
```

Comment:
Would it be good to rename `Unprotect ` to `UnprotectPeer` for more clarity?

Reply:
Same for `Protect`

Reply:
the functions have to be named what they are to satisfy the libp2p connection manager interface I believe